### PR TITLE
search for and link to MeSH descriptors.

### LINKED ISSUE #897: MeSH Search Results Limited to 21 Results
I noticed when doing a search for "Heart" that there was no way (in my estimation) that all of the results would have started with the letter "a".  See below ...

meshsrch2

So I did another search for "Brain" and got the following results ending with the letter "c" - "Central Nervous System Stimulants".

meshsrch

I decided to count the results and in both cases, I came up with a rough count of 21.  We should not be limiting MeSH results returned, non?

### PR DESCRIPTION
**Pull request by stopfstedt**
- [x] align styling with topics and co
- [x] fix saving of course/mesh and session/mesh association 
- [x] implement pagination or infinity scroll for mesh search.

refs #867 

fixes #897

### CODE REVIEW

**Comment by stopfstedt** [comment]
separating selected MeSH terms with commas

selection_015

**Comment by jrjohnson** [comment]
@stopfstedt is this ready for review?  It still has the WIP tag - so I'm just confirming.

**Comment by stopfstedt** [comment]
as requested by @saschaben - here's take two:

selection_016

**Comment by stopfstedt** [comment]
@jrjohnson this is ready to be reviewed.

File: app/templates/components/mesh-manager.hbs [hunk 1]
```diff
@@ -13,13 +13,16 @@
...
+  {{#if searchResults.length}}
     <div class='selectable-list'>
       <ul class='mesh-search-results'>
-        {{#each sortedSearchResults as |term|}}
+        {{#each searchResults as |term|}}
           <li class="{{unless term.isActive 'disabled'}}" {{action 'add' term}}>{{term.name}}</li>
         {{/each}}
       </ul>
+      {{#if hasMoreSearchResults }}
+          <button {{action 'searchMore' query}}>{{t 'general.showMore' }}</button>
```
Comment (jrjohnson) [inline app/templates/components/mesh-manager.hbs]: Why pass query here as opposed to just using this.get('query') inside of the searchMore action?

**Comment by jrjohnson** [comment]
:+1:  this can be merged as is.  My one comment is more for reference than an actual problem.

**Comment by stopfstedt** [comment]
@jrjohnson made changes according to your feedback, this should be GTG again.

**Comment by stopfstedt** [comment]
@homu +r

**Comment by jrjohnson** [comment]
:+1:

**Pull request merged**